components/courses/app/services/courses/student/total_hash.rb

```diff
@@ -5,7 +5,7 @@ class TotalHash < ApplicationService
       NAME    = 'name'
       LECTURE = 'lecture'
 
-      def initialize(students, sort_by, lecture_id)
+      def initialize(students, sort_by = nil, lecture_id = nil)
```

Comment:
What if I call `.new([Student.first], 3)`... what would be 3? sory_by or lecture_id?

Here is better to use named attributes: `def initialize(students, sort_by: nil, lecture_id: nil)`